admin/kernel-syncdata-service/src/main/java/io/mosip/kernel/syncdata/controller/SyncDataController.java

```diff
@@ -57,279 +74,412 @@ public class SyncDataController {
 	 * Service instance {@link SyncConfigDetailsService}
 	 */
 	@Autowired
-	SyncConfigDetailsService syncConfigDetailsService;
+	private SyncConfigDetailsService syncConfigDetailsService;
 
 	@Autowired
-	SyncUserDetailsService syncUserDetailsService;
+	private SyncUserDetailsService syncUserDetailsService;
 
 	@Autowired
-	LocalDateTimeUtil localDateTimeUtil;
+	private LocalDateTimeUtil localDateTimeUtil;
 
 	@Autowired
-	SyncJobHelperService syncJobHelperService;
-
+	private SyncJobHelperService syncJobHelperService;
 
-	
 	/**
-	 * 
-	 * @param keyIndex     - keyIndex mapped to machine
-	 * @param lastUpdated  - last sync updated time stamp
-	 * @param regCenterId  - regcenterId mapped to machine
-	 * @return {@link SyncDataResponseDto}
-	 * @throws InterruptedException - this method will throw interrupted Exception
-	 * @throws ExecutionException   - this method will throw exeution exception
+	 * Synchronizes client settings for a registration center and machine.
+	 *
+	 * @param keyIndex    the machine key index
+	 * @param lastUpdated the last updated timestamp (optional)
+	 * @param regCenterId the registration center ID (optional)
+	 * @return {@link ResponseWrapper} containing {@link SyncDataResponseDto}
+	 * @throws Throwable if data fetching fails
 	 */
 	@PreAuthorize("hasAnyRole(@authorizedRoles.getGetclientsettings())")
 	@ResponseFilter
+	@ApiOperation(value = "API to sync client settings")
 	@GetMapping("/clientsettings")
 	public ResponseWrapper<SyncDataResponseDto> syncClientSettings(
-			@RequestParam(value = "keyindex", required = true) String keyIndex,
-			@RequestParam(value = "lastUpdated", required = false) String lastUpdated,
-			@RequestParam(value = "regcenterId", required = false) String regCenterId)
+			@ApiParam("Machine key index") @RequestParam(value = "keyindex", required = true) String keyIndex,
+			@ApiParam("Last updated timestamp") @RequestParam(value = "lastUpdated", required = false) String lastUpdated,
+			@ApiParam("Registration center ID") @RequestParam(value = "regcenterId", required = false) String regCenterId)
 			throws Throwable {
+		MDC.put(KEY_INDEX_KEY, keyIndex);
+		try {
+			LOGGER.debug("syncClientSettings invoked with keyIndex: {}, regCenterId: {}, lastUpdated: {}",
+					keyIndex.replaceAll("[\n\r]", "_"), regCenterId, lastUpdated);
```

Comment:
Can we instead replace with empty string. And I think we should do sanitize all 3 parameters

Reply:
Removed the debug logger